__tests__/index.test.ts

```diff
@@ -633,10 +640,86 @@ describe('cast', () => {
   })
 
   test('casts binary text data to text', () => {
-    expect(cast({ name: 'test', type: 'VARBINARY', flags: 4225 }, 'table')).toEqual('table')
+    expect(cast({ name: 'test', type: 'VARBINARY', charset: 255 }, 'table')).toEqual('table')
   })
 
   test('casts JSON string to JSON object', () => {
     expect(cast({ name: 'test', type: 'JSON' }, '{ "foo": "bar" }')).toStrictEqual({ foo: 'bar' })
   })
 })
+
+describe('parse e2e', () => {
+  test('golden test', async () => {
+    const mockResponse = {
+      session: mockSession,
+      result: JSON.parse(
```

Comment:
If I were smarter, could we read this from disk? And read the `golden/result-compact.json` file instead?